### PR TITLE
Fix bug in using inherited destructors

### PR DESCRIPTION
**Pull request by lydia-duncan**
I discovered that if a class used its parent destructor, we'd encounter an
internal assertion error, because for some reason we didn't expect to insert
coercion temps.  This was a really dumb assertion, made from pure laziness, so
I removed it and wrapped the code in a block that could be removed so that what
the assertion was worried about didn't come to pass either (as suggested by
@hildeth and the comment on the assertion).

Add a test to ensure this case will continue to work.

### CODE REVIEW

**Comment by lydia-duncan** [comment]
@hildeth, want to review?  I'm testing now.

**Comment by lydia-duncan** [comment]
Passed full testing

**Comment by lydia-duncan** [comment]
The Travis failure has to do with downloading a package for the chpldoc build and is not due to my change.  Though having a wrong hash for downloading is a little worrisome on its own . . .

File: test/classes/lydia/useInheritedDestructor.chpl [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    delete uninit;
+  }
+}
+
+class B: A {
+  var count: int = 1;
+}
+
+var instantiated: B = new B();
+writeln(instantiated);
```
Comment (hildeth) [inline test/classes/lydia/useInheritedDestructor.chpl]: I recommend freeing "instantiated" here.

Comment (lydia-duncan) [inline test/classes/lydia/useInheritedDestructor.chpl]: Heh, yeah . . .

**Comment by hildeth** [comment]
Fine to commit, after adding delete(instantiated); to the test program.

**Pull request merged**